lib/nostrum/voice.ex

```diff
@@ -834,4 +846,34 @@ defmodule Nostrum.Voice do
   end
 
   defp persistent_args(_voice), do: []
+
+  unless Application.compile_env(:nostrum, :suppress_youtubedl_version, false) do
+    @ytdl_old_version "2021.12.17"
+
+    defp check_youtubedl_version do
+      with ytdl when is_binary(ytdl) <- youtubedl_executable(),
+           ["youtube-dl" | _] <- ytdl |> Path.basename() |> String.split("."),
+           {version, 0} <- System.cmd(ytdl, ["--version"]),
+           version when version <= @ytdl_old_version <- String.trim(version) do
+        Logger.warning("""
+        Configured youtube-dl installation '#{System.find_executable(ytdl)}' is version #{version}.
+
+        This version no longer compatible with recent YouTube updates.
+
+        It is recommended to configure :nostrum, :youtubedl to point to a maintained fork such as 'yt-dlp'.
+
+        If you know what you are doing, configure :nostrum, :suppress_youtubedl_version to `true` to suppress this.
+        """)
+      end
+    end
+
+    defp maybe_warn(:ytdl) do
+      unless :persistent_term.get(:has_checked_youtubedl_version, false) do
+        :persistent_term.put(:has_checked_youtubedl_version, true)
+        {:ok, _pid} = Task.start(fn -> check_youtubedl_version() end)
+      end
+    end
```

Comment:
We should namespace this:
```suggestion
    defp maybe_warn(:ytdl) do
      unless :persistent_term.get(:nostrum_has_checked_youtubedl_version, false) do
        :persistent_term.put(:nostrum_has_checked_youtubedl_version, true)
        {:ok, _pid} = Task.start(fn -> check_youtubedl_version() end)
      end
    end
```

Reply:
👌